shared/src/components/tabulator/DateTimePickerEditor.vue

```diff
@@ -0,0 +1,135 @@
+      :clearable="false"
+      :confirm="true"
+      :open.sync="typeEditorActive"
+      ref="datepicker"
+      v-model="datePickerValue"
+      prefix-class="bkdates"
+      confirm-text="click to confirm new selection"
+      @confirm="submit"
+    />
+    <i
+      class="material-icons special-type"
+      @mousedown.prevent.stop="toggleTypeEditor"
+      :title="this.typeEditorTitle"
+    >{{ this.typeEditorIcon }}</i>
+  </NullableInputEditor>
+</template>
+<script lang="ts">
+/*
+  https://github.com/mengxiong10/vue2-datepicker. There is an update for vue 3 when we're ready for that https://github.com/mengxiong10/vue-datepicker-next
+*/
+import _ from 'lodash'
+import Vue from 'vue'
+import DatePicker from 'vue2-datepicker'
+import helpers from '@shared/lib/tabulator'
+import NullableInputEditor from './NullableInputEditor.vue'
+export default Vue.extend({
+  props: ['cell', 'params'],
+  components: { DatePicker, NullableInputEditor },
+  data() {
+    return {
+      datePickerValue: null,
+      rendered: false,
+      typeEditorActive: false
+    }
+  },
+  computed: {
+    typeEditorTitle() {
+      if (this.typeEditorActive) return 'Use standard input'
+      return 'Open date/time picker' 
+    },
+    typeEditorIcon() {
+      if (this.typeEditorActive) return 'edit'
+      return 'edit_calendar' 
+    },
+    datePickerType() {
+      const dataType = this.params?.dataType?.trim().toLowerCase() ?? ''
+
+      if (dataType === 'date' || dataType === 'daterange') {
+        return 'date'
+      }
+
+      if (this.isTimeType(dataType)) {
+        return 'time'
+      }
+
+      return 'datetime'
+    }
+  },
+  methods: {
+    isTimeType(dataValue) {
+      const times = [
+        'time',
+        'timetz',
+        'time without time zone',
+        'time with time zone'
+      ]
+
+      return times.includes(dataValue.trim().toLowerCase().replace(/ *\([^)]*\) */g, ""))
+    },
+    toggleTypeEditor() {
+      this.typeEditorActive = !this.typeEditorActive
+    },
+    submit(e) {
+      // if the typeEditor is active and you blur, don't submit anything
+      if (e.type === 'blur' && this.typeEditorActive) return false
+
+      this.$emit('value', this.datePickerValue)
+
+      this.toggleTypeEditor = false
+    },
+    nullifyInput() {
+      this.toggleTypeEditor = false
+      this.$emit('value', null)
+    }
+  },
+  watch: {
+    rendered() {
+      this.$set(this.$refs.nullable.$data, 'rendered', this.rendered)
+    },
+    typeEditorActive() {
+      if (this.typeEditorActive) {
+        const refValue = this.cell.getValue()
+        const dataType = this.params.dataType || ''
+        let dataValue = refValue == null ? refValue : helpers.niceString(refValue)
+
+        if (this.isTimeType(dataType) && dataValue !== null) {
+          dataValue = dataValue.search(/(\+|-)/i) > -1 && !isNaN(dataValue.slice(-1)) ? `${dataValue}:00`: dataValue  
+          this.datePickerValue = new Date(`2023-03-31T${dataValue}`)
```

Comment:
Where did this date come from? Should this be today's date?

Reply:
If not, or it doesn't matter, please add a comment or something

Reply:
It shouldn't matter what the date is since we are really just creating a date object to grab the time, it was a Parks and Recreations reference. There are no meetings on Mar 31st because it's not a real day.